Tests/GitHubReferences.Tests.ps1

```diff
@@ -0,0 +1,201 @@
+                }
+            }
+
+            $null = Remove-GitHubRepository -OwnerName $ownerName -RepositoryName $repositoryName
+        }
+
+        Describe 'Getting a reference(branch) from repository' {
+            $repositoryName = [Guid]::NewGuid()
+            $repo = New-GitHubRepository -RepositoryName $repositoryName -AutoInit
+            $masterRefName = "heads/master"
+            $randomRefName = "heads/someRandomRef"
+
+            Context 'On getting a valid reference from a new repository' {
+                $reference = Get-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName -Reference $masterRefName
+
+                It 'Should return details of the reference' {
+                    $reference.ref | Should be "refs/$masterRefName"
+                }
+            }
+
+            Context 'On getting an invalid reference from a new repository' {
+                $reference = Get-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName -Reference $randomRefName
+
+                It 'Should not return any details' {
+                    $reference | Should be $null
+                }
+            }
+
+            Context 'On getting a valid reference using Uri from a new repository' {
+                $reference = Get-GitHubReference -Uri $repo.svn_url -Reference $masterRefName
+
+                It 'Should return details of the reference' {
+                    $reference.ref | Should be "refs/$masterRefName"
+                }
+            }
+
+            Context 'On getting an invalid reference using Uri from a new repository' {
+                $reference = Get-GitHubReference -Uri $repo.svn_url -Reference $randomRefName
+
+                It 'Should not return any details' {
+                    $reference | Should be $null
+                }
+            }
+
+            $null = Remove-GitHubRepository -OwnerName $ownerName -RepositoryName $repositoryName
+        }
+
+        Describe 'Getting all references from repository' {
+            $repositoryName = [Guid]::NewGuid()
+            $repo = New-GitHubRepository -RepositoryName $repositoryName -AutoInit
+            $masterRefName = "heads/master"
+            $secondRefName = "heads/branch1"
+            $existingref = Get-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName -Reference $masterRefName
+            $sha = $existingref.object.sha
+            New-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName -Reference $secondRefName -Sha $sha
+            $refNames = @("refs/$masterRefName", "refs/$secondRefName")
+
+            Context 'On getting all references from a new repository' {
+                $reference = Get-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName
+
+                It 'Should return all references' {
+                    ($reference.ref | Where-Object {$refNames -Contains $_}).Count | Should be $refNames.Count
```

Comment:
This felt like a little too much programming for a test case, but I wasn't happy with checking for the count alone either. Should I just manually check if both items are present or is this fine?

Reply:
I think this is fine.  Thanks for checking.